taropsbt/interface.go

```diff
@@ -0,0 +1,213 @@
+	p.Inputs[index].assetProof = proof
+}
+
+// HasSplitCommitment determines if this transaction results in an asset split.
+// This is either the case if the value of the input asset is split or if one of
+// the outputs is non-interactive (in which case we need to have a zero value
+// tombstone asset in the change output).
+func (p *VPacket) HasSplitCommitment() (bool, error) {
+	for i := 0; i < len(p.Outputs); i++ {
+		// If any of the recipient asset is nil, this virtual
+		// transaction hasn't been prepared correctly, and we cannot
+		// determine if there is a split commitment.
+		if p.Outputs[i].Asset == nil {
+			return false, fmt.Errorf("recipient asset %d is nil", i)
+		}
+
+		// If there is a split commitment witness present, we know there
+		// is a split going on.
+		if p.Outputs[i].Asset.HasSplitCommitmentWitness() {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// Anchor is a struct that contains all the information about an anchor output.
+type Anchor struct {
+	// Value is output value of the anchor output.
+	Value btcutil.Amount
+
+	// PkScript is the output script of the anchor output.
+	PkScript []byte
+
+	// SigHashType is the signature hash type that should be used to sign
+	// the anchor output spend.
+	SigHashType txscript.SigHashType
+
+	// InternalKey is the internal key of the anchor output that the input
+	// is spending the asset from.
+	InternalKey *btcec.PublicKey
+
+	// MerkleRoot is the root of the tap script merkle tree that also
+	// contains the Taro commitment of the anchor output.
+	MerkleRoot []byte
+
+	// TapscriptSibling is the tapscript sibling of the Taro commitment.
+	TapscriptSibling []byte
+
+	// Bip32Derivation is the BIP32 derivation of the anchor output's
+	// internal key.
+	//
+	// TODO(guggero): Do we also want to allow multiple derivations to be
+	// specified here? That would allow us to specify multiple keys involved
+	// in MuSig2 for example. Same for the Taproot derivation below.
+	Bip32Derivation *psbt.Bip32Derivation
+
+	// TrBip32Derivation is the Taproot BIP32 derivation of the anchor
+	// output's internal key.
+	TrBip32Derivation *psbt.TaprootBip32Derivation
+}
+
+// VInput represents an input to a virtual asset state transition transaction.
+type VInput struct {
+	// PInput is the embedded default PSBT input struct that is used for
+	// asset related input data.
+	psbt.PInput
+
+	// PrevID is the asset previous ID of the asset being spent.
+	PrevID asset.PrevID
+
+	// Anchor contains the information about the BTC level anchor
+	// transaction that committed to the asset being spent.
+	Anchor Anchor
+
+	// asset is the full instance of the asset being spent. It is not
+	// exported because the assets script key must be encoded in the PSBT
+	// input struct for the signing to work correctly.
+	asset *asset.Asset
+
+	// assetProof is the proof that the asset being spent was committed to
+	// in the anchor transaction above.
+	assetProof *commitment.Proof
```

Comment:
Is this useful if none of the header or tapscript information is also included? 

My inclusion proof earlier, meant the actual full input proof: https://github.com/lightninglabs/taro/blob/main/proof/proof.go#L30-L69

Note that this "just" the full inclusion proof for the previous input, and not total asset history. A few of these fields are already present here like the transaction, and merkle proof, etc. 

Perhaps we useful middle ground would be a `InclusionProof() *proof.Proof` method that assembles this struct for callers given the information in this PSBT?

Reply:
I wasn't sure if we wanted to have all the duplicated data (like `PrevOut`, `AnchorTx` and `Asset`) with the full proof. And it also creates an issue with a circular package dependency if we use `*proof.Proof`. But I see how it can be useful, so I opted to use a byte slice instead of the structured `*proof.Proof`.

Reply:
I'll add a commit to verify the proof before signing as well as using the input proof when creating the receiver proofs in part 3 tomorrow.